text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,386 @@
+- Feature Name: cargo_crates_sigstore_integration
+- Start Date: 2023-03-27
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This document proposes the integration with Sigstore for cargo and crates.io to support signing and verifying crates.
+
+Sigstore consists of a set of components that together provide a complete artifact signing and verification mechanism. Sigstore was created as a “common good” service much in the same way that Let’s Encrypt was and in fact shares some underlying service components with Let’s Encrypt, most notably the Transparency Log implementation in the Rekor service. Moreover, Sigstore addresses some of the inherent challenges in legacy signing mechanisms such as the necessity to have long term sensitive key management capabilities.
+
+This document will address two primary use case scenarios:
+
+1. The developer building Crates locally on their desktop
+1. Crates produced through a formal build plant (GitHub Actions, CircleCI, etc. and commercial enterprise internal build plants)
```

Comment:
I maintain a Sigstore client implementation, and it isn't a term I've seen used in the Sigstore ecosystem before. 

@lulf my suggestion would be to use a taxonomy of "account identities" and "machine identities" instead here; that's closer to what most Sigstore clients use, and doesn't imply that the machine identity is actually a "builder" (which it doesn't have to be) 🙂

Reply:
I'm a passerby. `build plant` may be an unfortunate mis-translation for `build farm` for ESL speakers (e.g., as in power plant). I second the use of machine/builder identity instead :)

Reply:
Yes. It could also be "build factory". Which I have seen, though less commonly! Technically you can call a factory a "plant", as in "manufacturing plant", but... it's not really as common due to the meaning collision. I only mentioned it because it seems like "build plants" was trying to be a commonly-used colloquialism and... whiffed.

I have also encountered "artifact producer" and "artifact repository", used in more general senses, but also "build bots", "build machines", and "builder nodes".

Reply:
"build system" is a generic term that can cover quite a few bases